core/src/consistency.rs

```diff
@@ -51,6 +33,26 @@ impl ConsistencySignal {
     }
 }
 
+impl From<ConsistencySignalE> for ConsistencySignal<String> {
+    fn from(signal: ConsistencySignalE) -> ConsistencySignal<String> {
+        let ConsistencySignalE { event, pending } = signal;
+        ConsistencySignal {
+            event: serde_json::to_string(&event)
+                .expect("ConsistencySignal serialization cannot fail"),
+            pending: pending
+                .into_iter()
+                .map(|p| PendingConsistency {
+                    event: serde_json::to_string(&p.event)
+                        .expect("ConsistencySignal serialization cannot fail"),
+                    group: p.group,
+                })
+                .collect(),
+        }
+    }
+}
```

Comment:
This is the main new part, most of the rest of the PR is just moving stuff around.